File: kafka/src/test/groovy/io/micronaut/configuration/kafka/docs/producer/fallback/MessageClientFallback.java
package io.micronaut.configuration.kafka.docs.producer.fallback;

// tag::imports[]
import io.micronaut.context.annotation.Replaces;
import io.micronaut.context.annotation.Requires;
import io.micronaut.core.util.StringUtils;
// end::imports[]

// tag::clazz[]
@Requires(property = "kafka.enabled", notEquals = StringUtils.TRUE, defaultValue = StringUtils.TRUE) // <1>
@Replaces(MessageClient.class) // <2>
public class MessageClientFallback implements MessageClient { // <3>

    @Override
    public void send(String message) {
        throw new UnsupportedOperationException(); // <4>
    }
}
// end::clazz[]

File: test-suite-python/src/test/python/micronaut/kafka/docs/producer/fallback/MessageClientFallback.py
# tag::imports[]
from jakarta.inject import Singleton
from micronaut.context.annotation import Replaces, Requires
from micronaut.core.util import StringUtils

from .MessageClient import MessageClient
# end::imports[]


@Requires(property="spec.name", value="MessageClientFallbackTest")
# tag::clazz[]
@Requires(property="kafka.enabled", notEquals=StringUtils.TRUE, defaultValue=StringUtils.TRUE)  # <1>
@Replaces(MessageClient)  # <2>
@Singleton
class MessageClientFallback(MessageClient):  # <3>

    def send(self, message: str) -> None:
        raise NotImplementedError()  # <4>
# end::clazz[]
